src/lava/magma/core/model/nc/model.py

```diff
@@ -38,3 +106,148 @@ def allocate(self, net: Net):
         Note: This should work as before.
         """
         pass
+
+
+class NcProcessModel(AbstractNcProcessModel):
+    def __init__(self, proc_params: ty.Dict[str, ty.Any],
+                 board: ty.Type[NxBoard],
+                 loglevel=logging.WARNING):
+        super(AbstractNcProcessModel, self).__init__(proc_params,
+                                                     loglevel=loglevel)
+        self.board = board
+
+    def start(self):
+        self.service_to_process.start()
+        self.process_to_service.start()
+        for p in self.nc_ports:
+            p.start()
+
+        self.run()
+
+    def allocate(self):
+        pass
+
+    def run(self):
+        """Retrieves commands from the runtime service calls
+        their corresponding methods of the ProcessModels. The phase
+        is retrieved from runtime service (service_to_process). After
+        calling the method of a phase of all ProcessModels the runtime
+        service is informed about completion. The loop ends when the
+        STOP command is received."""
+        selector = CspSelector()
+        channel_actions = [(self.service_to_process, lambda: 'cmd')]
+        action = 'cmd'
+        while True:
+            if action == 'cmd':
+                cmd = self.service_to_process.recv()
+                if enum_equal(cmd, MGMT_COMMAND.STOP):
+                    self.board.stop()
```

Comment:
There might be multiple NcProcessModels. As every process model will get mgmt command and every one tries to relay it to board - wouldn't this be a problem. I would suggest moving all the board mgmt logic to NxSDKRuntimeService.

Reply:
> There might be multiple NcProcessModels. As every process model will get mgmt command and every one tries to relay it to board - wouldn't this be a problem. I would suggest moving all the board mgmt logic to NxSDKRuntimeService.

ACK, no problem. I asked about this in a chat with you but perhaps I misunderstood or didn't communicate what I meant:
"[2/11/2022 4:44 PM] Marcus
Is it a problem if all of the process models have a reference to the same board?"
[2/11/2022 4:45 PM] Joyesh
They must have reference to the same board. There is a single board at the moment.
I mean there is no problem"